spec/support/preload_values.rb

```diff
@@ -0,0 +1,33 @@
+# Example usage:
+#   expect { MyModel.do_the_queries }.to preload_values(:attribute, value)
+
+RSpec::Matchers.define :preload_values do |field, expected_values|
+  match(:notify_expectation_failures => true) do |block|
+    results = block.call.order(:id).load
+
+    @count = ActiveRecord::QueryCounter.count do
+      if results.respond_to?(:map)
+        expected_values = Array(expected_values) unless expected_values.kind_of?(Array)
+        actual = results.map(&field.to_sym.to_proc)
+        expect(actual).to eq(expected_values)
+      else
+        expect(results.send(field)).to eq(expected_values)
+      end
+    end
+    @count == 0
+  end
+
+  failure_message do |_actual|
+    "Expected to preload values (#{@count} queries)"
```

Comment:
I feel like this might look weird, because you are testing two things:

- that the values are what we expect
- the query count is `0` (after the `.load` is called above)

Can you provide some output for failure cases for each of the above?  Do they seem readable?

Reply:
since we use `notify_expectation_failures` the inner expect will report it's own failure.
So this failure is actually just when the record count does not work.

Reply:
> "Expected to preload #{@field} but executed #{@count} queries instead"

I'll look at these failure cases